### ISSUE TITLE
Phase 3: RSI Bot Setup and execution

### ISSUE THREAD

**Issue by SattikMohanta** [SattikMohanta/Trading-Bot-with-the-Intraday-Historical-Data-API-from-EODHD---NYSE-Stock#18]
Creates a plot with two subplots: one for the closing price and another for the RSI.

    Parameters:
        dataframe (pd.DataFrame): DataFrame containing stock data with a 'close' column.
        rsi (pd.Series): Series containing RSI values.
        figsize (tuple): Size of the figure.

    Returns:
        tuple: A tuple containing the Figure and a list of Axes objects.

**Comment by SattikMohanta** [comment]
# Plot Closing Price
    axs[0].plot(dataframe.index, dataframe['close'], label='Close Price', color='blue', linewidth=1.5)
    axs[0].set_title("Closing Price", fontsize=14, fontweight='bold')
    axs[0].set_ylabel("Price (USD)", fontsize=12)
    axs[0].legend()
    axs[0].grid(True, linestyle="--", alpha=0.6)

    # Plot RSI
    axs[1].plot(rsi.index, rsi, label='RSI', color='orange', linewidth=1.5)
    axs[1].axhline(70, linestyle='--', color='red', alpha=0.5)  # Overbought line
    axs[1].axhline(30, linestyle='--', color='green', alpha=0.5)  # Oversold line
    axs[1].set_title("Relative Strength Index (RSI)", fontsize=14, fontweight='bold')
    axs[1].set_ylabel("RSI", fontsize=12)
    axs[1].legend()
    axs[1].grid(True, linestyle="--", alpha=0.6)

    # Improve x-axis visibility
    axs[1].tick_params(axis="x", rotation=45)  # Rotate x-axis labels for better visibility

    plt.xlabel("Date", fontsize=12)
    
    return fig, axs

**Comment by SattikMohanta** [comment]
# Configuration
symbol = "NVDA.US"
interval = "10m"
range_days = "90d"
api_key = "67d6200c2f87e5.90891934"   # Replace with your actual API Key
from_date_str = '01-01-2025'     # MM-DD-Y (yyyy)
to_date_str   = '31-03-2025'     # MM-DD-Y (yyyy)

url = f"https://eodhistoricaldata.com/api/eod/NVDA.US?api_token=67d6200c2f87e5.90891934&from=01-01-2025&to=31-03-2025&fmt=json"

# Fetch historical/eod data
response = requests.get(url)
data = response.json()
df = pd.DataFrame(data)

# Convert and sort data
df['date'] = pd.to_datetime(df['date'])
df.sort_values('date', inplace=True)
df.set_index('date', inplace=True)

# Calculate RSI
delta = df['close'].diff()
gain = delta.clip(lower=0)
loss = -delta.clip(upper=0)
avg_gain = gain.rolling(RSI_PERIOD).mean()
avg_loss = loss.rolling(RSI_PERIOD).mean()
rs = avg_gain / avg_loss
df['RSI'] = 100 - (100 / (1 + rs))

# Plot RSI with Plotly
fig = go.Figure()

# RSI Line
fig.add_trace(go.Scatter(
    x=df.index,
    y=df['RSI'],
    mode='lines',
    name='RSI',
    line=dict(color='dodgerblue', width=2)
))

# Overbought line (70)
fig.add_trace(go.Scatter(
    x=[df.index[0], df.index[-1]],
    y=[70, 70],
    mode='lines',
    name='Overbought (70)',
    line=dict(color='red', width=1, dash='dash'),
    showlegend=True
))

# Oversold line (30)
fig.add_trace(go.Scatter(
    x=[df.index[0], df.index[-1]],
    y=[30, 30],
    mode='lines',
    name='Oversold (30)',
    line=dict(color='green', width=1, dash='dash'),
    showlegend=True
))

# Text annotations
fig.add_annotation(x=df.index[-1], y=70, text="Overbought", showarrow=False, font=dict(color="red"))
fig.add_annotation(x=df.index[-1], y=30, text="Oversold", showarrow=False, font=dict(color="green"))

# Layout
fig.update_layout(
    title=f"RSI Chart for {symbol}",
    xaxis_title="Date",
    yaxis_title="RSI",
    legend=dict(x=0.01, y=0.99, bgcolor="rgba(0,0,0,0)", font=dict(size=10)),
    template="simple_white",
    height=500
)

fig.show()

**Comment by SattikMohanta** [comment]
![Image](https://github.com/user-attachments/assets/8fde878e-3b97-4896-9b2a-c455e3766c84)

**Comment by SattikMohanta** [comment]
# Configuration
symbol = "NVDA.US"
interval = "15m"
range_days = "90d"
api_key = "67d6200c2f87e5.90891934"   # Replace with your actual API Key
from_date_str = '01-01-2025'     # MM-DD-Y (yyyy)
to_date_str   = '31-03-2025'     # MM-DD-Y (yyyy)

url = f"https://eodhistoricaldata.com/api/eod/NVDA.US?api_token=67d6200c2f87e5.90891934&from=01-01-2025&to=31-03-2025&fmt=json"
# Fetch data
response = requests.get(url)
data = response.json()

# Check structure
print("Sample record from response:")
print(data[:5]) # Preview first 5 sliced entry

**Comment by SattikMohanta** [comment]
# Convert to DataFrame
df = pd.DataFrame(data)
df['date'] = pd.to_datetime(df['date'])  # Correct column name
df.set_index('date', inplace=True)
df.sort_index(inplace=True)


# Plot candlestick chart
fig = go.Figure(data=[go.Candlestick(
    x=df.index,
    open=df['open'],
    high=df['high'],
    low=df['low'],
    close=df['close'],
    increasing_line_color='green',
    decreasing_line_color='red'
)])

fig.update_layout(
    title=f'{symbol} Intraday Candlestick Chart ({interval})',
    xaxis_title='Time',
    yaxis_title='Price',
    xaxis_rangeslider_visible=False,
    template='seaborn',
    height=600
)

fig.show()

**Comment by SattikMohanta** [comment]
![Image](https://github.com/user-attachments/assets/2f163b3a-3971-4369-a605-8af2a7163373)

**Comment by SattikMohanta** [comment]
# Configuration
symbol = "NVDA.US"
interval = "30m"
range_days = "90d"
api_key = "67d6200c2f87e5.90891934"   # Replace with your actual API Key
from_date_str = '01-01-2025'     # MM-DD-Y (yyyy)
to_date_str   = '31-03-2025'     # MM-DD-Y (yyyy)

url = f"https://eodhd.com/api/eod/NVDA.US?interval=5m&range=1d&api_token=67d6200c2f87e5.90891934&fmt=json"
# Fetch data
response = requests.get(url)
data = response.json()

# === API REQUEST ===
url = f"https://eodhistoricaldata.com/api/eod/NVDA.US?api_token=67d6200c2f87e5.90891934&from=01-01-2025&to=31-03-2025&fmt=json"
response = requests.get(url)

# === ERROR HANDLING ===
if response.status_code != 200:
    raise Exception(f"API request failed with status {response.status_code}: {response.text}")

# === PARSE JSON TO DATAFRAME ===
data = response.json()
df = pd.DataFrame(data)

df['date'] = pd.to_datetime(df['date'])
df.set_index('date', inplace=True)
df.sort_index(inplace=True)


# === GROUP BY WEEK ===
df['week'] = df.index.to_period('W')
weekly_close = df.groupby('week')['close'].apply(list)

# === PREPARE DATA FOR BOXPLOT ===
box_data = weekly_close.tolist()
week_labels = [str(week) for week in weekly_close.index]

# === PLOT BOXPLOT ===
plt.figure(figsize=(14, 6))
box = plt.boxplot(box_data, patch_artist=True)

# === COLOR STYLING ===
colors = ['lightblue', 'lightgreen', 'lightcoral', 'wheat', 'lavender']
for patch, color in zip(box['boxes'], colors * (len(box['boxes']) // len(colors) + 1)):
    patch.set_facecolor(color)

plt.title(f'Weekly Closing Price Distribution for {symbol}', fontsize=16)
plt.xlabel('Week')
plt.ylabel('Closing Price')
plt.xticks(ticks=range(1, len(week_labels)+1), labels=week_labels, rotation=45, fontsize=9)
plt.grid(axis='y', linestyle='--', alpha=0.7)
plt.tight_layout()
plt.show()
